### DISCUSSION TITLE
Best editor for Jupyter Books?

### DISCUSSION THREAD

**Discussion by NatalieZelenka** [executablebooks/meta#121] [Q&A] [answered]
This is an inconsequential problem, really, but it's winding me up:

I'm saving my Jupyter Book pages as markdown files, but the markdown support in my usual editor (PyCharm - I use the free version) is confused by directives. It doesn't recognise the first set of backticks in a directive as being related to a comment since it's immediately followed by the curly brackets, so it greys out everything in between the end of the first directive and the second, then the end of the third and the fourth, etc. 

Does anyone have a favourite editor for writing Jupyter Books?

**Comment by chrisjsewell** [comment, marked as answer]
Heya see: https://jupyterbook.org/content/myst.html#vscode
Also see checkout the new “markdown friendly” syntaxes in https://jupyterbook.org/reference/_changelog.html#new-and-improved 😀

**Reply by chrisjsewell** [reply]
Basically I’d ditch PyCharm for VS Code 😉, because these days it’s just a lot better

**Reply by chrisjsewell** [reply]
And please let me know if any improvements can be made 👍

**Reply by NatalieZelenka** [reply]
Great, I'll give it a go, thanks!

**Reply by oscarbranson** [reply]
But the inability to add tags to notebook cells in VS Code is rather limiting...

**Comment by fm75** [comment]
I have been using `binder` (not the public mybinder.org version) with JupyterLab and the Show Markdown Preview. Unfortunately, it does not render things like the "special content blocks" like Notes, Warnings, Admonitions, ... the same. The same seems to be true when using that extended markdown in a notebook.

Commit, push changes back. Pull to local repo and `make build`, `make publish`. My github instance is a GHE version that does not yet support Actions.

**Reply by chrisjsewell** [reply]
Well I guess it depends what you mean by "the same".
Naturally there is no Markdown previewer than can fully replicate how jupyter-book/sphinx actually renders the notebooks (cross-refences, directives, styling, ...), unless we were to literally rewrite jupyter-book/sphinx in JavaScript 😬.
But the point of the extended markdown is that it makes it at least a bit more comparable:

```note
Writing a big admonition like this, which does **not** render *any* Markdown, or render things like [links](https://www.example.com), is really not great.
```

:::{note}
But, even though this does not give you the full JB rendering, this is **at least** *better*, and works across any Markdown render, with things like [links](https://www.example.com) etc
:::

With things like https://github.com/executablebooks/myst-language-support and https://github.com/executablebooks/markdown-it-myst though, we are looking to move towards better Javacript renderers.
Then with the maturing of the Jupyter Lab infrastructure (with things like https://github.com/krassowski/jupyterlab-lsp), hopefully, we can integrate these into that, or create a plugin 😄 
(just got to find the time in the day lol)

**Comment by rowanc1** [comment]
Another option is to use [Curvenote](https://curvenote.com/), which is a technical wysiwyg editor designed for writing books/papers with computational content linked from Jupyter and speaks MyST markdown/jupyter-book. There are some pointers on how to export directly to jupyter-book using the CLI tools here:
https://cli.curvenote.dev/jupyter-book.html

This is especially helpful for working with collaborators who need to review work, but don't work with git/code/markdown/myst/etc.
(I am one of the authors behind Curvenote, and also helping to develop the javascript myst libraries behind the vscode plugins!)
